### DISCUSSION TITLE
Publish to OpenUPM through Github Releases

### DISCUSSION THREAD

**Discussion by PereViader** [openupm/openupm#4918] [Ideas]
When doing a release to OpenUPM right now, the process requires that the repository is already formatted as a package so it can be picked up and served by OpenUPM.

There is however no easy way to integrate OpenUPM and custom build processes.
The docs mention that new ideas should be given here :)

Projects could publish an artifact as GitHub Releases with whatever custom build process they need.
OpenUPM could consume artifacts (`.unitypackage` / `.tar.gz` / `.zip`) hosted within GitHub Releases of the repository.

Example:
- Push commit
- Github action runs tests
- Github action runs publishing script
  - Publishes relevant packages to nuget
  - Publishes relevant packages to Github Release
- OpenUPM detects a new package within Github Release and starts serving the new version

**Comment by PereViader** [comment]
Seems like my idea is not novel. I found another proposal 
https://github.com/openupm/openupm/discussions/1872#discussioncomment-5671079
https://github.com/openupm/openupm/issues/3923

**Reply by favoyang** [reply]
Fixed in https://github.com/openupm/openupm/issues/6402#issuecomment-4409311479
